File: backend/src/requests/add_feed.rs
use rusqlite::{Connection};
use protocol::request::AddFeed;
use protocol::Response;
use super::Error;

pub fn add_feed(request: AddFeed, conn: &mut Connection) -> Result<Response, Error> {
    let mut stmt = conn.prepare_cached("INSERT INTO feeds(url) VALUES ( ? )")?;

    stmt.execute(&[ &request.url ])?;

    Ok(Response::Success)
}

File: frontend/src/thunks/addFeed.ts
import Connection from "../Connection";
import { Dispatch } from 'redux';
import Action from "../actions";
import getFeeds from "./getFeeds";

async function addFeed(dispatch: Dispatch<Action>, conn: Connection, url: string) {
  await conn.addFeed(url);
  await getFeeds(dispatch, conn);
}
export default addFeed;
